[CheckpointWrapper] Test for kwarg propagation, remove checkpoint_fn_arg support

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #102679

Closes https://github.com/pytorch/pytorch/issues/100576

Differential Revision: [D46342398](https://our.internmc.facebook.com/intern/diff/D46342398/)

rohan-varma committed:
[CheckpointWrapper] Test for arg propagation

Closes https://github.com/pytorch/pytorch/issues/100576

Differential Revision: [D46342398](https://our.internmc.facebook.com/intern/diff/D46342398/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[CheckpointWrapper] Test for arg propagation"


Closes https://github.com/pytorch/pytorch/issues/100576

Differential Revision: [D46342398](https://our.internmc.facebook.com/intern/diff/D46342398/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[CheckpointWrapper] Test for arg propagation"


Closes https://github.com/pytorch/pytorch/issues/100576

Differential Revision: [D46342398](https://our.internmc.facebook.com/intern/diff/D46342398/)

[ghstack-poisoned]

awgu commented on test/distributed/fsdp/test_checkpoint_wrapper.py:
```diff
@@ -100,6 +101,33 @@ def forward(self, *, a=None, b=None):
+            torch.nn.Linear(1, 1),
+            checkpoint_fn=torch_utils_checkpoint,
+            use_reentrant=False,
+            context_fn=get_ctx_mgrs,
```
I am guessing this is the kwarg. Should we test arg as well?

rohan-varma replied:
Decided to eliminate `checkpoint_fn_args` support, as there's really no practical way to use this.

rohan-varma committed:
Update on "[CheckpointWrapper] Test for arg propagation"


Closes https://github.com/pytorch/pytorch/issues/100576

Differential Revision: [D46342398](https://our.internmc.facebook.com/intern/diff/D46342398/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[CheckpointWrapper] Test for kwarg propagation, remove checkpoint_fn_arg support"


Closes https://github.com/pytorch/pytorch/issues/100576

Differential Revision: [D46342398](https://our.internmc.facebook.com/intern/diff/D46342398/)

[ghstack-poisoned]

rohan-varma commented:
@pytorchbot merge -f "CI done"
